src/filters/renderer/VideoRenderers/MPCVRAllocatorPresenter.cpp

```diff
@@ -266,7 +270,7 @@ STDMETHODIMP CMPCVRAllocatorPresenter::CreateRenderer(IUnknown** ppRenderer)
         return E_UNEXPECTED;
     }
 
-    if (IsCLSIDRegistered(CLSID_MPCVR) || !LoadInternalMPCVRFilter()) {
+    if (IsCLSIDRegistered(CLSID_MPCVR) || !InstantiateInternalMPCVR(m_pMPCVR, GetOwner())) {
         HRESULT hr = m_pMPCVR.CoCreateInstance(CLSID_MPCVR, GetOwner());
```

Comment:
It's still needed when the first call is true.  Logic is a bit weird, because if the both are true, CoCreateInstance should fail anyway.

Reply:
Sorry, I mean if first call returns false, but second returns true (meaning it couldn't load internal), then it will try to create the instance, which I expect to fail since the CLSID is not registered.  But it doesn't hurt to try I suppose.

Reply:
See what you think about the updated code.  I think it's clearer than before.